scripts/machine_translation/train_transformer.py

```diff
@@ -422,36 +425,50 @@ def train(args):
     log_start_time = time.time()
     num_params, num_fixed_params = None, None
     # TODO(sxjscience) Add a log metric class
-    accum_count = 0
-    loss_denom = 0
-    n_train_iters = 0
-    log_wc = 0
-    log_avg_loss = 0.0
-    log_loss_denom = 0
-    epoch_id = 0
-    while (args.epochs < 0 or epoch_id < args.epochs): # when args.epochs < 0, the model will keep training
-        n_epoch_train_iters = 0
-        processed_batch_num = 0
-        train_multi_data_loader = grouper(train_data_loader, len(ctx_l))
-        is_last_batch = False
-        sample_data_l = next(train_multi_data_loader)
-        while not is_last_batch:
-            processed_batch_num += len(sample_data_l)
+    log_avg_loss_l = [mx.np.array(0.0, ctx=ctx) for ctx in ctx_l]
+    log_avg_loss_counter = 0
+    log_wc_l = [mx.np.array(0, dtype=np.int64, ctx=ctx) for ctx in ctx_l]
+
+    if local_rank == 0:
+        writer = SummaryWriter(logdir=os.path.join(args.save_dir, 'tensorboard'))
+    if use_amp:
+        amp.init_trainer(trainer)
```

Comment:
We should allow the users to set the initial scaling, e.g., 2^20.

Reply:
I checked the API and find that there is no such option in the amp in MXNet. Is it necessary for large batch?

Reply:
This is something necessary.

Reply:
And, the ability of tuning scaling factor, window size, etc is also needed.

Reply:
I see. What would be the typical flag in pytorch AMP training? I think the MXNet AMP has not offered the functionality of setting the options. Need to confirm with NVIDIA.